src/tsolvers/lasolver/SparseMatrix.cc

```diff
@@ -0,0 +1,168 @@
+/*
+ *  Copyright (c) 2022, Martin Blicha <martin.blicha@gmail.com>
+ *
+ *  SPDX-License-Identifier: MIT
+ *
+ */
+
+#include "SparseMatrix.h"
+
+
+void SparseColMatrix::Col::negate() {
+    this->poly.negate();
+}
+
+void SparseColMatrix::Col::add(Col const & other, opensmt::Real const & multiple) {
+    this->poly.merge(other.poly, multiple, [](LVRef){}, [](LVRef){});
+}
+
+opensmt::Real SparseColMatrix::Col::product(const std::vector<FastRational> & values) const {
+    opensmt::Real sum = 0;
+    for (auto const & term : poly) {
+        uint32_t index = term.var.x;
+        assert(index < values.size());
+        sum += term.coeff * values[index];
+    }
+    return sum;
+}
+
+SparseColMatrix::TermVec SparseColMatrix::Col::toVector() const {
```

Comment:
Not sure we need this `TermVec`, it's basically a restricted version of `Polynomial`.
I am a bit worried about the unnecessary copying, but it's probably not a big deal...

Reply:
As discussed, let's keep it as it is for now.